docs/preview/features/message-pumps/service-bus.md

```diff
@@ -302,6 +303,58 @@ string transactionId = correlationInfo.TransactionId;
 string operationId = correlationInfo.OperationId;
 ```
 
+## Automatic Azure Key Vault credentials rotation
+
+The additional library `Arcus.Messaging.Pumps.ServiceBus.KeyRotation` provides an extension on the message pump to restart the pump automatically when the credentials of the pump stored in Azure Key Vault are changed.
+This feature allows more reliable restarting instead of relying on authentication exceptions that may be throwed during the lifetime of the message pump.
+
+## How does this work?
+
+A background job will be run in the background which is in fact a message pump on a Service Bus Topic.
+The Azure Key Vault where the connection string for your target message pump is located should send `SecretNewVersionCreated` events to this Service Bus Topic.
+
+That way, when the background job receives a new Key Vault event, it will know that your target message pump should be restarted.
```

Comment:
```suggestion
That way, when the background job receives a new Key Vault event, it will get the latest connection string, restart the message pump and authenticate with the latest credentials.
```